File: server/app/models/upgrade.py
"""Upgrade (升级包) model."""
from __future__ import annotations

from sqlalchemy import BigInteger, String
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin


class Upgrade(Base, TimestampMixin):
    __tablename__ = "upgrades"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    model: Mapped[str] = mapped_column(String(64), default="")
    appVersion: Mapped[str] = mapped_column(String(32), default="")
    sortAppVersion: Mapped[int] = mapped_column(BigInteger, default=0)
    originname: Mapped[str] = mapped_column(String(256), default="")
    name: Mapped[str] = mapped_column(String(256), default="")
    path: Mapped[str] = mapped_column(String(512), default="")
    md5: Mapped[str] = mapped_column(String(64), default="")

File: smarttriage/lib/models/upgrade.js
'use strict';

module.exports = (Schema) => {
  /**
   * Upgrade（升级包）
   */
  const UpgradeSchema = new Schema({
    model: String, // model
    appVersion: String, // 版本号
    sortAppVersion: Number, // 版本号数值
    originname: String, // 原始文件名
    name: String, // 文件名
    path: String, // 路径
    md5: String, // md5
    ct: Date, // 创建时间
    ut: Date // 更新时间
  });

  return UpgradeSchema;
};
